io/src/main/scala/fs2/io/file/package.scala

```diff
@@ -0,0 +1,56 @@
+package fs2.io
+
+import java.nio.ByteBuffer
+import java.nio.channels.{AsynchronousFileChannel, CompletionHandler}
+import java.nio.file.{OpenOption, Path, StandardOpenOption}
+
+import fs2._
+import fs2.util.Monad
+
+package object file {
+
+  private[fs2] def asyncCompletionHandler[F[_], O](f: CompletionHandler[O, Null] => F[Unit])(implicit F: Async[F]): F[O] = {
+    F.async[O] { cb =>
+      f(new CompletionHandler[O, Null] {
+        override def completed(result: O, attachment: Null): Unit = cb(Right(result))
+
+        override def failed(exc: Throwable, attachment: Null): Unit = cb(Left(exc))
+      })
+    }
+  }
+
+  // uses default NIO thread pool for callbacks... okay? should be configurable?
+  def open[F[_]](path: Path, flags: Seq[OpenOption])(implicit F: Monad[F]): Pull[F, Nothing, AsynchronousFileChannel] = {
+    Stream.bracket(F.pure(AsynchronousFileChannel.open(path, flags: _*)))(
+      Stream.emit, { chan => F.suspend(chan.close()) }
+    ).open.flatMap(_.await1.map(_.head))
+  }
+
+  def readAsync[F[_]](chan: AsynchronousFileChannel, numBytes: Int, offset: Long)(implicit F: Async[F]): F[Option[Chunk[Byte]]] = {
+    val buf = ByteBuffer.allocate(numBytes)
+    F.bind(
+      asyncCompletionHandler[F, Integer](f => F.pure(chan.read(buf, offset, null, f))))(
+      len => F.pure {
+        if (len < 0) None else if (len == 0) Some(Chunk.empty) else Some(Chunk.bytes(buf.array.take(len)))
+      }
+    )
+  }
+
+  def readAllAsync[F[_]](path: Path, chunkSz: Int)(implicit F: Async[F]): Stream[F, Byte] = {
+    def readAllFromChannel(chunkSz: Int, offset: Long)(chan: AsynchronousFileChannel): Pull[F, Byte, Unit] = {
```

Comment:
`readAllFromChannel` may be useful as a top-level method (if folks acquire an `AsynchronousFileChannel` via a call to `open`.